src/clr-angular/forms/datepicker/monthpicker.spec.ts

```diff
@@ -0,0 +1,297 @@
+/*
+ * Copyright (c) 2016-2018 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import {Component} from "@angular/core";
+import {async} from "@angular/core/testing";
+
+import {TestContext} from "../../data/datagrid/helpers.spec";
+import {IfOpenService} from "../../utils/conditional/if-open.service";
+import {DOWN_ARROW, LEFT_ARROW, RIGHT_ARROW, UP_ARROW} from "../../utils/key-codes/key-codes";
+
+import {DayModel} from "./model/day.model";
+import {ClrMonthpicker} from "./monthpicker";
+import {DateIOService} from "./providers/date-io.service";
+import {DateNavigationService} from "./providers/date-navigation.service";
+import {DatepickerViewService} from "./providers/datepicker-view.service";
+import {LocaleHelperService} from "./providers/locale-helper.service";
+import {ViewManagerService} from "./providers/view-manager.service";
+import {createKeyboardEvent} from "./utils/test-utils";
+
+export default function() {
+    describe("Monthpicker Component", () => {
+        let context: TestContext<ClrMonthpicker, TestComponent>;
+        let localeHelperService: LocaleHelperService;
+        let dateNavigationService: DateNavigationService;
+        const selectedMonth: number = 1;
+
+        function initializeCalendar(selectedDay: DayModel) {
+            dateNavigationService = new DateNavigationService();
+            // Setting a selected date so that the calendar is initialized to that month and year.
+            dateNavigationService.selectedDay = selectedDay;
+            dateNavigationService.initializeCalendar();
+        }
+
+        describe("View Basics", () => {
+            beforeEach(function() {
+                initializeCalendar(new DayModel(2015, selectedMonth, 1));
+
+                context = this.create(ClrMonthpicker, TestComponent, [
+                    ViewManagerService, DatepickerViewService, IfOpenService,
+                    {provide: DateNavigationService, useValue: dateNavigationService}, LocaleHelperService,
+                    DateIOService
+                ]);
+                localeHelperService = context.getClarityProvider(LocaleHelperService);
+            });
+
+            it("projects the months correctly", () => {
```

Comment:
It's not projection.

Reply:
sorry. changed to `renders the month...`